### PR TITLE
[stable/mariadb] Add PodDisruptionBudget for MariaDB

### PR DESCRIPTION
**Pull request by tomasbedrich**
#### What this PR does / why we need it:
Adds option to enable [PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) for MariaDB master and slave StatefulSets.

#### Which issue this PR fixes
none

#### Special notes for your reviewer:
have a nice day :)

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/mariadb/templates/master-pdb.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  minAvailable: {{ .Values.master.podDisruptionBudget.minAvailable }}
+{{- end }}
+{{- if .Values.master.podDisruptionBudget.maxUnavailable }}
+  maxUnavailable: {{ .Values.master.podDisruptionBudget.maxUnavailable }}
+{{- end }}
+  selector:
+    matchLabels:
+      app: "{{ template "mariadb.name" . }}"
+      component: "master"
+      release: {{ .Release.Name | quote }}
```
Comment (juan131) [inline stable/mariadb/templates/master-pdb.yaml:23]: For consistency with the `metadata.labels.release`, use:
```
release: {{ .Release.Name | quote }}
```

File: stable/mariadb/templates/slave-pdb.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  minAvailable: {{ .Values.slave.podDisruptionBudget.minAvailable }}
+{{- end }}
+{{- if .Values.slave.podDisruptionBudget.maxUnavailable }}
+  maxUnavailable: {{ .Values.slave.podDisruptionBudget.maxUnavailable }}
+{{- end }}
+  selector:
+    matchLabels:
+      app: "{{ template "mariadb.name" . }}"
+      component: "slave"
+      release: {{ .Release.Name | quote }}
```
Comment (juan131) [inline stable/mariadb/templates/slave-pdb.yaml:24]: For consistency with the `metadata.labels.release`, use:
```
release: {{ .Release.Name | quote }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks so much for this contribution @tomasbedrich  and sorry for the late response.

A couple of requests:
- Add the same changes you added to `values.yaml` to `values-production.yaml`
- Rebase from master to fix the conflicts.

**Comment by tomasbedrich** [comment]
- [x] Add the same changes you added to `values.yaml` to `values-production.yaml`
- [x] Rebase from master to fix the conflicts.

According to quoting – I presume that you intended to comment on line 22 (`app: "{{ template "mariadb.name" . }}"`), not line 24 (`release...
release: {{ .Release.Name | quote }}`) as I was already using `quote` function on release name. So I unified all PDB quoting to use `quote` function – please review.

File: stable/mariadb/templates/master-pdb.yaml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{{- if .Values.master.podDisruptionBudget.enabled }}
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  name: {{ template "mariadb.fullname" . }}
+  labels:
+    app: {{ template "mariadb.name" . | quote }}
```
Comment (juan131) [inline stable/mariadb/templates/master-pdb.yaml]: Revert this change. It doesn't work

```
Error: render error in "mariadb/templates/slave-pdb.yaml": template: mariadb/templates/_helpers.tpl:6:18: executing "mariadb.name" at <.Chart.Name>: can't evaluate field Chart in type string
```

**Review by juan131 — CHANGES_REQUESTED** [review]

File: stable/mariadb/templates/master-pdb.yaml (hunk 3 above)
Comment (juan131) [reply stable/mariadb/templates/master-pdb.yaml]: I tested it using the command below:

```
$ helm template --name mariadb stable/mariadb --set master.podDisruptionBudget.enabled=true,slave.podDisruptionBudget.enabled=true
```

Comment (tomasbedrich) [reply stable/mariadb/templates/master-pdb.yaml]: Oops, will revert. So can you please clarify your previous note?

> For consistency with the `metadata.labels.release`, use: `release: {{ .Release.Name | quote }}`

**Review by juan131 — APPROVED** [review]
Thanks!! Approving your changes.
